src/main/java/net/fabricmc/loom/configuration/ide/RunConfigSettings.java

```diff
@@ -0,0 +1,279 @@
+	 */
+	private final List<String> programArgs = new ArrayList<>();
+
+	/**
+	 * The mode to run, which is the name of the run config in {@code fabric_installer.[method].json}.
+	 */
+	private String mode;
+
+	/**
+	 * The full name of the run configuration, i.e. 'Minecraft Client'.
+	 *
+	 * <p>By default this is determined from the base name.
+	 */
+	private String name;
+
+	/**
+	 * Whether the run config is for the client, i.e. uses natives and needs {@code -XstartOnFirstThread} on OSX.
+	 */
+	private Boolean client;
+
+	/**
+	 * The default main class of the run configuration.
+	 *
+	 * <p>This can be overwritten in {@code fabric_installer.[method].json}. Note that this <em>doesn't</em> take
+	 * priority over the main class specified in the Fabric installer configuration.
+	 */
+	private String defaultMainClass;
+
+	/**
+	 * The source set getter, which obtains the source set from the given project.
+	 */
+	private Function<Project, SourceSet> source;
+
+	/**
+	 * The run directory for this configuration, relative to the root project directory.
+	 */
+	private String runDir;
+
+	/**
+	 * The base name of the run configuration, which is the name it is created with, i.e. 'client'
+	 */
+	private final String baseName;
+
+	private final Project project;
+	private final LoomGradleExtension extension;
+
+	public RunConfigSettings(Project project, String baseName) {
+		this.baseName = baseName;
+		this.project = project;
+		this.extension = project.getExtensions().getByType(LoomGradleExtension.class);
+
+		mode(baseName);
+		source("main");
+		runDir("run");
+	}
+
+	public Project getProject() {
+		return project;
+	}
+
+	public LoomGradleExtension getExtension() {
+		return extension;
+	}
+
+	@Override
+	public String getName() {
+		return baseName;
+	}
+
+	public List<String> getVmArgs() {
+		return vmArgs;
+	}
+
+	public List<String> getProgramArgs() {
+		return programArgs;
+	}
+
+	public String getMode() {
+		return mode;
+	}
+
+	public void setMode(String mode) {
+		this.mode = mode;
+	}
+
+	public String getConfigName() {
+		return name;
+	}
+
+	public void setConfigName(String name) {
+		this.name = name;
+	}
+
+	public String getDefaultMainClass() {
+		return defaultMainClass;
+	}
+
+	public void setDefaultMainClass(String defaultMainClass) {
+		this.defaultMainClass = defaultMainClass;
+	}
+
+	public String getRunDir() {
+		return runDir;
+	}
+
+	public void setRunDir(String runDir) {
+		this.runDir = runDir;
+	}
+
+	public SourceSet getSource(Project proj) {
+		return source.apply(proj);
+	}
+
+	public void setSource(SourceSet source) {
+		this.source =
+				proj -> source;
+	}
+
+	public void setSource(Function<Project, SourceSet> sourceFn) {
+		this.source = sourceFn;
+	}
+
+	public void mode(String mode) {
+		setMode(mode);
+	}
+
+	public void name(String name) {
+		setConfigName(name);
+	}
+
+	public void defaultMainClass(String cls) {
+		setDefaultMainClass(cls);
+	}
+
+	public void runDir(String dir) {
+		setRunDir(dir);
```

Comment:
No point of such `set` overloads. Dunno how it looks in kotlin but in groovy it is difference between `runDir "run"` (this one) and `runDir = "run"` (setter), and I don't think it's worth creating an overload for an `=` sign

Reply:
just `get/set` should suffice, kotlin will generate a `val` accessor in compiler and groovy will treat it like a property.